actors/miner/src/lib.rs

```diff
@@ -1054,7 +1054,7 @@ impl Actor {
                 let quant = state.quant_spec_for_deadline(rt.policy(), dl_idx);
 
                 for update in &decls_by_deadline[&dl_idx] {
-                    rt.verify_replica_update(&update.proof_inputs).with_context_code(
+                    Verifier::verify_replica_update(rt, &update.proof_inputs).with_context_code(
```

Comment:
I don't think requiring this calling convention will be reasonable given the method is available directly on rt. I also can't tell why the change is needed though. If there is a real thing happening here, I think it's too subtle for future developers to follow.

Reply:
Since runtime implements both Primitives and Verifier which both now have `verify_replica_update` with the same signature, we need to specify which method to dispatch to. 

The Primitives could have differently named functions - perhaps prefixed?

Reply:
> Since runtime implements both Primitives and Verifier which both now have verify_replica_update with the same signature

Ok I think the design problem originates here. Primitives and Verifier are IMO just a boring and unnecessary separation of the methods required by a complete Runtime impl. It shouldn't drive any other code structure. We could in principle just inline them all into Runtime, and perhaps should in practise if the separation cause problems. We'd still then need a trait for external behaviour to inject into a VM, and forwarding methods. But the Primitives trait is doing two jobs here, and perhaps should do just one.

- One solution might be to keep the separation and have Primitives and Verifier separately injectable into the VM. I.e. _not_ "Define all injectable behaviour as a method of the Primitives trait".
- Another is to delete the Verifier trait and merge its methods into Primitives. The distinction is pretty arbitrary anyway. There might be some friction around the `fake-proofs` build config though.